src/lambda-handler.js

```
@@ -0,0 +1,69 @@
+import env from './decrypt-env';
+
+export default class LambdaHandler {
+  /* BEGIN LIFECYCLE METHODS */
+  validateEvent(event) {
+    return event;
+  }
+
+  async preRun(event, context) {
+  }
+
+  async run(event, context) {
+    throw new Error('Run method must be implemented in handler subclass');
+  }
+
+  async postRun(event, context) {
+  }
+  /* END LIFECYCLE METHODS */
+
+  _processEvent(event) {
+    return event;
+  }
+
+  _processResponse(result) {
+    return result;
+  }
+
+  _processError(err) {
+    return err;
+  }
+
+  async _execute(event, context) {
+    try {
+      await env.decrypt();
+
+      event = this._processEvent(event);
+      event = this.validateEvent(event);
```

If `this.validateEvent` does not have a return value, we should throw an error so that developers know they have to return something here.  Otherwise `event` would be `undefined` in all subsequent lifecycle methods, which would probably throw an exception somewhere anyway.